### ISSUE TITLE
Please avoid different versions for the same source

### ISSUE THREAD

**Issue by gasinvein** [lapce/lapce#61]
In Linux distributions, Cargo vendoring feature is often used to bundle dependencies, for caching on CI, ensuring reproducibility and doing builds offline.
But `cargo vendor` will fail if the same source used in a project has different versions.
In Lapse case, `xi-rope` package has different versions. If possible, please try to avoid this. It'll help Linux packagers a big deal.

_From a comment in another issue:_
> > Failed to build:
> > ```
> > lapce> error: failed to sync
> > lapce> Caused by:
> > lapce>   found duplicate version of package `xi-rope v0.3.0` vendored from two sources:
> > lapce>          source 1: registry `crates-io`
> > lapce>          source 2: https://github.com/lapce/xi-editor#5578c725
> > lapce> Traceback (most recent call last):
> > lapce>   File "/nix/store/qcr9904iv7flbkd9g56rk5fc11lyn7df-cargo-vendor-normalise/bin/.cargo-vendor-normalise-wrapped", line 42, in <module>
> > lapce>     main()
> > lapce>   File "/nix/store/qcr9904iv7flbkd9g56rk5fc11lyn7df-cargo-vendor-normalise/bin/.cargo-vendor-normalise-wrapped", line 17, in main
> > lapce>     assert list(data.keys()) == ["source"]
> > lapce> AssertionError
> > ```
> 
> Seems like nix can't handle multiple versions of the same crate with the same version string. Try building using cargo directly instead of nix.

The problem is not nix, the error is thrown by `cargo`, the error comes out even before the build starts while getting `vendor` dependencies.
cargo can process any number of the same crate from both different versions and from different sources, provided that the code is written correctly in the target application, more than once I've met such applications where one crate had different sources and versions.

If we run `cargo vendor` to get vendored sources we get the same error
```
❯ cargo vendor --manifest-path Cargo.toml  
error: failed to sync

Caused by:
  found duplicate version of package `xi-rope v0.3.0` vendored from two sources:

  	source 1: registry `crates-io`
  	source 2: https://github.com/lapce/xi-editor#5578c725
```
This shouldn't be
Rust 1.57.0

_Originally posted by @DeadMozay in https://github.com/lapce/lapce/issues/3#issuecomment-994913767_

**Comment by dzhou121** [comment]
Should be fixed in v0.0.7

**Comment by gasinvein** [comment]
Yep, seems resolved now, thanks.
